core/templates/dev/head/components/top_navigation_bar/TopNavigationBarDirective.js

```diff
@@ -95,13 +95,91 @@ oppia.directive('topNavigationBar', [function() {
         }
 
         $scope.windowIsNarrow = windowDimensionsService.isWindowNarrow();
+        var currentWindowWidth = windowDimensionsService.getWidth();
+        $scope.navElementsVisibilityStatus = {
+          I18N_TOPNAV_DONATE: true,
+          I18N_TOPNAV_ABOUT: true,
+          I18N_CREATE_EXPLORATION_CREATE: true,
+          I18N_TOPNAV_LIBRARY: true
+        };
+
         windowDimensionsService.registerOnResizeHook(function() {
           $scope.windowIsNarrow = windowDimensionsService.isWindowNarrow();
           $scope.$apply();
           // Close the sidebar, if necessary.
           SidebarStatusService.closeSidebar();
+
+          // If the window is resized larger, try displaying the hidden elements
+          if (currentWindowWidth < windowDimensionsService.getWidth()) {
+            for (element in $scope.navElementsVisibilityStatus) {
+              if (!$scope.navElementsVisibilityStatus[element]) {
+                $scope.navElementsVisibilityStatus[element] = true;
+              }
+            }
+          }
+          currentWindowWidth = windowDimensionsService.getWidth();
+          truncateNavbarDebounced();
         });
 
+        /**
+         * Checks if i18n has been run.
+         * If i18n has not yet run, the <a> and <span> tags will have
+         * no text content, so their innerText.length value will be 0.
+         * @returns {boolean}
+         */
+        var checkIfI18NCompleted = function() {
+          var i18nCompleted = true;
+          document.querySelectorAll('.oppia-navbar-tabs a[translate], ' +
+            '.oppia-navbar-tabs span[translate]').forEach(function(element) {
+            if (element.innerText.length === 0) {
+              i18nCompleted = false;
+              return false;
+            }
+          });
+          return i18nCompleted;
+        };
+
+        /**
+         * Checks if window is >768px and i18n is completed, then checks
+         * for overflow. If overflow is detected hides the least important
+         * tab and then calls itself again after a 50ms delay.
+         */
+        var truncateNavbar = function() {
+          // If the window is narrow, the standard nav tabs are not shown.
+          if (windowDimensionsService.isWindowNarrow()) {
+            return false;
+          }
+
+          // If i18n hasn't completed, retry after 100ms.
+          if (!checkIfI18NCompleted()) {
+            $timeout(truncateNavbar, 100);
+            return false;
+          }
+
+          // Measured non-overflowed navbar height under 60px via inspector.
```

Comment:
OK, please add this elaboration in the code about how the 60 was obtained. I think it is useful.